src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs

```diff
@@ -159,13 +156,16 @@ public async Task Invoke(HttpContext context)
                     WriteDiagnosticEvent(_diagnosticSource, eventName, new { httpContext = context, exception = ex });
                 }
 
+                _metrics.RequestException(exceptionName, ExceptionResult.Unhandled, handler: null);
```

Comment:
Perhaps we need more thought into what is meant by Handled and Unhandled.

I put `Unhandled` here because it's literally right below a diagnostic event that reports the exception as unhandled: https://github.com/dotnet/aspnetcore/blob/6575b289081b9f85fb30d01c7b306d1ec5d00a68/src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs#L156-L160